portal-impl/test/unit/com/liferay/portal/util/FileImplTest.java

```diff
@@ -223,6 +236,134 @@ public void testStripSuffixWhenFileNameHasParenthesisAtStart() {
 			"()test.jsp", _fileImpl.stripParentheticalSuffix("()test.jsp"));
 	}
 
+	@Test
+	public void testUnzip() throws Exception {
+		_setUpForUnzipTests();
+
+		File file = _createZipFile(
+			"test.zip",
+			HashMapBuilder.put(
+				"zip/test/directory/", () -> (String)null
+			).put(
+				"zip/test/entry/entry.txt", "Test String"
+			).build());
+
+		_fileImpl.unzip(file, _tempFolder);
+
+		_assertExists("zip/test/directory");
+		_assertExists("zip/test/entry/entry.txt");
+
+		Assert.assertTrue(
+			_canFileReadWrite(_getTempTestFilePath("zip/test/directory")));
+
+		Assert.assertTrue(
+			_canFileReadWrite(
+				_getTempTestFilePath("zip/test/entry/entry.txt")));
+		_tearDown();
+	}
+
+	@Test
+	public void testUnzipZipSlipVulnerable() throws Exception {
+		_setUpForUnzipTests();
+
+		File file = _createZipFile(
+			"test_slip.zip",
+			HashMapBuilder.put(
+				"../../../../../../bad.txt", "I am bad!"
+			).put(
+				"good.txt", "I am good!"
+			).build());
+
+		_fileImpl.unzip(file, _tempFolder);
+
+		_assertExists("good.txt");
+		_assertDoesNotExist("tmp/bad.txt");
+		_tearDown();
+	}
+
+	private void _assertDoesNotExist(String name) throws Exception {
+		Path fullPath = _getTempTestFilePath(name);
+
+		Assert.assertFalse(Files.exists(fullPath));
+	}
+
+	private void _assertExists(String name) throws Exception {
+		Path fullPath = _getTempTestFilePath(name);
+
+		Assert.assertTrue(Files.exists(fullPath));
+	}
+
+	private boolean _canFileReadWrite(Path path) {
```

Comment:
@jeffwu0724 Why is this needed and asserted? By the way, why does it contain different logic for Windows and other OS?

Reply:
For the OS checking, in the beginning I am wondering there will be different path format for different OS. I will remove it now, and I will check it with `ci:test:environment` and see if we need this part or not.